tools/misc/browse_dataset.py

```diff
@@ -35,72 +38,171 @@ def parse_args():
     ]
 
     return cfg
 
 
+def to_depth_mode(points, bboxes):
+    """Convert points and bboxes to Depth Coord and Depth Box mode."""
+    if points is not None:
+        points = Coord3DMode.convert_point(points.copy(), Coord3DMode.LIDAR,
+                                           Coord3DMode.DEPTH)
+    if bboxes is not None:
+        bboxes = Box3DMode.convert(bboxes.clone(), Box3DMode.LIDAR,
+                                   Box3DMode.DEPTH)
+    return points, bboxes
+
+
+def show_det_data(idx, dataset, out_dir, filename):
+    """Visualize 3D point cloud and 3D bboxes."""
+    example = dataset.prepare_train_data(idx)
+    points = example['points']._data.numpy()
+    gt_bboxes = dataset.get_ann_info(idx)['gt_bboxes_3d'].tensor
+    if dataset.box_mode_3d != Box3DMode.DEPTH:
+        points, gt_bboxes = to_depth_mode(points, gt_bboxes)
+    show_result(
+        points,
+        gt_bboxes.clone(),
+        None,
+        out_dir,
+        filename,
+        show=True,
+        snapshot=True)
+
+
+def show_seg_data(idx, dataset, out_dir, filename):
+    """Visualize 3D point cloud and segmentation mask."""
+    example = dataset.prepare_train_data(idx)
+    points = example['points']._data.numpy()
+    gt_seg = example['pts_semantic_mask']._data.numpy()
+    show_seg_result(
+        points,
+        gt_seg.copy(),
+        None,
+        out_dir,
+        filename,
+        np.array(dataset.PALETTE),
+        dataset.ignore_index,
+        show=True,
+        snapshot=True)
+
+
+def show_proj_bbox_img(idx, dataset, out_dir, filename):
+    """Visualize 3D bboxes on 2D image by projection."""
+    example = dataset.prepare_train_data(idx)
+    gt_bboxes = dataset.get_ann_info(idx)['gt_bboxes_3d']
+    img_metas = example['img_metas']._data
+    img = example['img']._data.numpy()
+    # need to transpose channel to first dim
+    img = img.transpose(1, 2, 0)
+    # no 3D gt bboxes, just show img
+    if gt_bboxes.tensor.shape[0] == 0:
+        gt_bboxes = None
+    if isinstance(gt_bboxes, DepthInstance3DBoxes):
+        show_multi_modality_result(
+            img,
+            gt_bboxes,
+            None,
+            example['calib'],
+            out_dir,
+            filename,
+            depth_bbox=True,
+            img_metas=img_metas,
+            show=True)
+    elif isinstance(gt_bboxes, LiDARInstance3DBoxes):
+        show_multi_modality_result(
+            img,
+            gt_bboxes,
+            None,
+            img_metas['lidar2img'],
+            out_dir,
+            filename,
+            depth_bbox=False,
+            img_metas=img_metas,
+            show=True)
+    else:
+        # can't project, just show img
+        show_multi_modality_result(
+            img, None, None, None, out_dir, filename, show=True)
+
+
+def is_multi_modality(dataset):
+    """Judge whether a dataset loads multi-modality data (points+img)."""
+    if not hasattr(dataset, 'modality') or dataset.modality is None:
```

Comment:
Maybe we can remind the user to set the modality and what will be the default way to deal with this input

Reply:
Yes, I add an args. But I think we still need to check if this dataset really supports img+points loading (add a warning).